### PR TITLE
[0.x] Adds `Moderations::create` typed response

### PR DESCRIPTION
**Pull request by gehrisandro**
There we go.

see #4

### CODE REVIEW

**Comment by nunomaduro** [comment]
@gehrisandro It's possible to focus this pull request in **only on typed responses**, and on a single resource method?

**Comment by gehrisandro** [comment]
@nunomaduro I narrowed the pull request down to focus only on the typed response as you suggested.

In my opinion the typed response is very easy to understand and use.
But still there are some points I wasn't sure what the best approach would be:

- Is it a good idea to rearrange the data returned from the API or should the structure represented in exactly the same way? For the moderation endpoint, I rearranged the resulting categories data  from two arrays having the same keys to a single array with category objects, which feels for me more natural.

- The response objects are namespaced under "DataObjects" and not just simple "Responses". In my opinion it expresses a bit better, that it isn't a 100% exact copy of what the API returned. (only if you agree to rearrange data as mentioned above)

- Naming of the nested data objects: Moderation\ModerationCategory vs. Moderation\Category
Probably it's an endless discussion which approach is better. In my opinion most important is to be consistent within the project. What do you prefer?

- To create the data objects from the array I've created dedicated factory classes. Maybe it's a bit over engineered but it keeps the API call method every clean as it can simply call the factory with the data returned from the API.

- Last but not least. I had to add a `@phpstan-ignore-line` in ModerationFactory::make() as I wasn't able to find a proper type declaration which doesn't lead to an error.

I'm looking forward to hearing your opinion.

File: src/DataObjects/Moderation/Moderation.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace OpenAI\DataObjects\Moderation;
```
Comment (nunomaduro) [inline src/DataObjects/Moderation/Moderation.php:3]: This should be `OpenAI\Responses\Moderations;`.

File: src/DataObjects/Moderation/Moderation.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+use OpenAI\Contracts\DataObject;
+
+final class Moderation implements DataObject
```
Comment (nunomaduro) [inline src/DataObjects/Moderation/Moderation.php:7]: This class should be name `CreateResponse` and it should implement the `Response` interface.

File: src/DataObjects/Moderation/ModerationCategory.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace OpenAI\DataObjects\Moderation;
```
Comment (nunomaduro) [inline src/DataObjects/Moderation/ModerationCategory.php:3]: OpenAI\Responses\Moderations;

File: src/DataObjects/Moderation/ModerationCategory.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+use OpenAI\Contracts\DataObject;
+use OpenAI\Enums\Moderation\Category;
+
+final class ModerationCategory implements DataObject
```
Comment (nunomaduro) [inline src/DataObjects/Moderation/ModerationCategory.php:8]: This one should be named CreateResponseModerationCategory, etc.

**Comment by nunomaduro** [comment]
@gehrisandro Left a few comments on the way I would probably tackled this.

**Comment by gehrisandro** [comment]
Thank you for our feedback.

I renamed classes / namespaces in the way to suggested, updated and cleaned update the tests.

**Comment by nunomaduro** [comment]
Adjusting things locally.

**Pull request merged**